addons/website_forum/models/forum_forum.py

```diff
@@ -130,6 +130,19 @@ def _get_default_welcome_message(self):
     karma_user_bio = fields.Integer(string='Display detailed user biography', default=750)
     karma_post = fields.Integer(string='Ask questions without validation', default=100)
     karma_moderate = fields.Integer(string='Moderate posts', default=1000)
+    has_pending_post = fields.Boolean(string='Has pending post', compute="_compute_has_pending_post")
+
+    @api.depends_context('uid')
+    def _compute_has_pending_post(self):
+        for forum in self:
+            has_pending_post = self.env['forum.post'].search([
+                            ('create_uid', '=', self.env.user.id),
+                            ('state', '=', 'pending'),
+                            ('parent_id', '=', False),
+                            ('forum_id', '=', forum.id)
+                        ])
+
+            forum.has_pending_post = len(has_pending_post) > 0
```

Comment:
This is not ideal, because you will perform a SQL query for each iteration of the loop.
What we do instead is either
* use `read_group` so we have one request for all forum records, than process it as necessary, or better (when possible which is often the case),
* let the orm do the work of fetching the required data. When you loop over the records in `self`, the ORM will fetch the required data once for all records. It would mean here to filter the posts of the current user using `forum.post_ids.filtered` or `filtered_domain`.
I think however that loading all posts of a user simply to know if there are any pending may not be efficient, I'd go for `read_group`.
I'm available for further discussion ofc.